reflect_test.go

```diff
@@ -331,3 +336,19 @@ func TestBaselineUnmarshal(t *testing.T) {
 
 	require.Equal(t, strings.ReplaceAll(string(expectedJSON), `\/`, "/"), string(actualJSON))
 }
+
+func TestSplitOnUnescapedCommas(t *testing.T) {
+	strToSplit := "Hello,this,is\\,a\\,string,haha"
+	expected := []string{"Hello", "this", "is,a,string", "haha"}
+	actual := splitOnUnescapedCommas(strToSplit)
+
+	require.Equal(t, expected, actual)
+}
+
+func TestSplitOnUnescapedCommasDoubleEscaped(t *testing.T) {
```

Comment:
It's a bit more efficient to use an example table with a struct so that it is easier to add more examples. I'd suggest a single `TestSplitOnUnescapedCommas` that loops through a list of examples with expected output.

Reply:
That's a great suggestion, thanks! I just pushed a commit implementing your idea.